### ISSUE TITLE
document IRQ handling internals

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#559]
**_Reported by Andrew Boie:_**

I sent out a draft document explaining how interrupts work on X86:

https://lists.zephyrproject.org/archives/list/devel@lists.zephyrproject.org/thread/3OCS4M326UERBVV3AXDJ6437QYOCN6NW/

Clean this up, convert to ReST format, and commit to our in-tree documentation. Document how sw_isr_table works on ARC/ARM/Nios II as well.

(Imported from Jira ZEP-634)

**Comment by nashif** [comment]
**_by Anas Nashif:_**

is this WIP and will be in 1.6? please Accept issue in this case. if not, please move to 1.7

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Andrew Boie_ , please see Anas' comment above and act on this story accordingly.  Thanks.

**Comment by nashif** [comment]
**_by Andrew Boie:_**

Blocked by GH-1056 as that will involve a major revision to the IRQ implementation on ARC/ARM/NIOS

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Andrew Boie_ , any update on this?  Thx.

**Comment by nashif** [comment]
**_by Andrew Boie:_**

This is on hold as a major rewrite of IRQs is in progress for non-x86 arches see GH-1056, GH-942

**Comment by nashif** [comment]
**_by Andrew Boie:_**

_Mark Linkmeyer_  can we move this to 1.8? If I write this now, work on GH-1056 and GH-942 will make it already obsolete?

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Anas Nashif_ , please answer Andrew's question above about whether this can moved out to 1.8.  I assume it can because it's a Medium priority, but I'd like you to make the call.  Thx.  _Andrew Boie_ , I'm thinking there's no harm in leaving it as targeted for 1.7 since it's a nice-to-have anyhow.  If you can get it done in time, then good.  If not, then it will automatically get pushed to 1.8.  However, if you're 100% sure at this time that it can't be done in time for 1.7, then I see no issue with pushing it out now to 1.8, if Anas agrees.

**Comment by nashif** [comment]
**_by Andrew Boie:_**

I can start working on it once I get those other two JIRAs done.
Since this is documentation I think I could send it out even past the merge window closing.

**Comment by nashif** [comment]
**_by Anas Nashif:_**

_Andrew Boie_  yes, this documentation story can be submitted after merge window. so that is ok

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

Fixing incorrect priority

**Comment by nashif** [comment]
**_by David Kinder:_**

_Andrew Boie_  I can handle transcoding an approved version of the document you want into ReST -- is there something you can send me (or attach to this Jira)?

**Comment by nashif** [comment]
**_by Andrew Boie:_**

Not working on this -- preempted by runtime.io MCUBoot support and planning memory management stories for 1.8.
Suggest punting this to "Future" given how short-handed we are at the moment.

**Comment by nashif** [comment]
**_by Andrew Boie:_**

I have a draft up in a personal branch:
https://github.com/andrewboie/zephyr/tree/irq-docs

However I'm blocked due to GH-2020.

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Andrew Boie_  and _Anas Nashif_ , since this is a documentation story I assume it can still go into 1.8 this week, if it's ready.  Will it be ready?  It would be nice to get this one off our radar, if possible.

**Comment by nashif** [comment]
**_by Andrew Boie:_**

It's in review and I got some feedback from David, will have a new patch set up tomorrow.
